lib/astutils.cpp

```diff
@@ -376,7 +385,10 @@ bool isOppositeCond(bool isNot, bool cpp, const Token * const cond1, const Token
             (comp1 == ">"  && comp2 == "<=") ||
             (comp1 == ">=" && comp2 == "<") ||
             (!isNot && ((comp1 == "<" && comp2 == ">") ||
-                        (comp1 == ">" && comp2 == "<"))));
+                        (comp1 == ">" && comp2 == "<") ||
```

Comment:
It seems to me that these changes in `isOppositeCond()` are the only important changes. Can some of the other changes be skipped for now?

Reply:
The check for the condition needs to be updated to catch the cases in the issue because `isNot` is never false. Changing `isNot` to false for `&&` causes other test cases to fail. So I moved this check further down so I could reuse the logic to decide whether its floating point or inconclusive.